src/main/java/uk/gov/hmcts/ccd/domain/service/getcasedocument/CaseDocumentAmApiClient.java

```diff
@@ -1,67 +1,69 @@
+import feign.FeignException;
+import feign.FeignException.FeignClientException;
 import org.springframework.http.HttpHeaders;
-import org.springframework.http.HttpMethod;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.MediaType;
-import org.springframework.web.client.HttpClientErrorException;
-import org.springframework.web.client.RestTemplate;
-import uk.gov.hmcts.ccd.ApplicationParams;
 import uk.gov.hmcts.ccd.data.SecurityUtils;
 import uk.gov.hmcts.ccd.domain.model.search.CaseDocumentsMetadata;
 import uk.gov.hmcts.ccd.endpoint.exceptions.BadSearchRequest;
 import uk.gov.hmcts.ccd.endpoint.exceptions.DocumentTokenException;
 import uk.gov.hmcts.ccd.endpoint.exceptions.ResourceNotFoundException;
 import uk.gov.hmcts.ccd.endpoint.exceptions.ServiceException;
+import uk.gov.hmcts.reform.ccd.document.am.feign.CaseDocumentClientApi;
 
 import javax.inject.Inject;
 import javax.inject.Named;
 
+import java.util.List;
+
+import static uk.gov.hmcts.ccd.data.SecurityUtils.SERVICE_AUTHORIZATION;
+
 @Named
 public class CaseDocumentAmApiClient {
-    private final RestTemplate restTemplate;
+    private final CaseDocumentClientApi caseDocumentClientApi;
     private final SecurityUtils securityUtils;
-    private final ApplicationParams applicationParams;
+    private final CaseDocumentMetadataMapper caseDocumentMetadataMapper;
 
     @Inject
-    public CaseDocumentAmApiClient(final RestTemplate restTemplate,
+    public CaseDocumentAmApiClient(final CaseDocumentClientApi caseDocumentClientApi,
                                    final SecurityUtils securityUtils,
-                                   final ApplicationParams applicationParams) {
-        this.restTemplate = restTemplate;
+                                   final CaseDocumentMetadataMapper caseDocumentMetadataMapper) {
+        this.caseDocumentClientApi = caseDocumentClientApi;
         this.securityUtils = securityUtils;
-        this.applicationParams = applicationParams;
+        this.caseDocumentMetadataMapper = caseDocumentMetadataMapper;
     }
 
     public void applyPatch(final CaseDocumentsMetadata caseDocumentsMetadata) {
         final HttpHeaders headers = securityUtils.authorizationHeaders();
-        headers.add(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
-        final HttpEntity<CaseDocumentsMetadata> requestEntity = new HttpEntity<>(caseDocumentsMetadata, headers);
+
+        uk.gov.hmcts.reform.ccd.document.am.model.CaseDocumentsMetadata mappedCaseDocumentsMetadata =
+            caseDocumentMetadataMapper.convertToAmClientCaseDocumentsMetadata(caseDocumentsMetadata);
+
+        List<String> authHeaders = headers.get(HttpHeaders.AUTHORIZATION);
+        String authorization = authHeaders == null ? null : authHeaders.get(0);
+
+        List<String> serviceAuthHeaders = headers.get(SERVICE_AUTHORIZATION);
+        String serviceAuthorization = serviceAuthHeaders == null ? null : serviceAuthHeaders.get(0);
```

Comment:
We shouldn't get to this point without calling service being authorized ? Belt and braces NPE check to please Sonar?

Reply:
Fixed in 70dce10